cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/FullModel.java

```diff
@@ -0,0 +1,208 @@
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+
+            XMLStreamReader xmlReader = factory.createXMLStreamReader(reader);
+            try {
+                XmlUtil.readUntilStartElement(new String[] {"/", CgmesNames.RDF, CgmesNames.FULL_MODEL}, xmlReader, () -> {
+                    context.id = xmlReader.getAttributeValue(CgmesNamespace.RDF_NAMESPACE, CgmesNames.ABOUT);
+                    XmlUtil.readUntilEndElement(CgmesNames.FULL_MODEL, xmlReader, () -> {
+                        switch (xmlReader.getLocalName()) {
+                            case CgmesNames.SCENARIO_TIME:
+                                context.scenarioTime = ZonedDateTime.parse(XmlUtil.readUntilEndElement(CgmesNames.SCENARIO_TIME, xmlReader, () -> { }));
+                                break;
+                            case CgmesNames.CREATED:
+                                context.created = ZonedDateTime.parse(XmlUtil.readUntilEndElement(CgmesNames.CREATED, xmlReader, () -> { }));
+                                break;
+                            case CgmesNames.DESCRIPTION:
+                                context.description = XmlUtil.readUntilEndElement(CgmesNames.DESCRIPTION, xmlReader, () -> { });
+                                break;
+                            case CgmesNames.VERSION:
+                                context.version = Integer.parseInt(XmlUtil.readUntilEndElement(CgmesNames.VERSION, xmlReader, () -> { }));
+                                break;
+                            case CgmesNames.PROFILE:
+                                context.profiles.add(XmlUtil.readUntilEndElement(CgmesNames.PROFILE, xmlReader, () -> { }));
+                                break;
+                            case CgmesNames.DEPENDENT_ON:
+                                context.dependentOn.add(xmlReader.getAttributeValue(CgmesNamespace.RDF_NAMESPACE, CgmesNames.RESOURCE));
+                                break;
+                            case CgmesNames.SUPERSEDES:
+                                context.supersedes.add(xmlReader.getAttributeValue(CgmesNamespace.RDF_NAMESPACE, CgmesNames.RESOURCE));
+                                break;
+                            case CgmesNames.MODELING_AUTHORITY_SET:
+                                context.modelingAuthoritySet = XmlUtil.readUntilEndElement(CgmesNames.MODELING_AUTHORITY_SET, xmlReader, () -> { });
+                                break;
+                            default:
+                                // not yet interesting like superseded
+                                break;
+                        }
+                    });
+                });
+            } finally {
+                xmlReader.close();
+            }
+        } catch (XMLStreamException e) {
+            throw new UncheckedXmlStreamException(e);
+        }
+        context.validate();
```

Comment:
I would merge the validate method and the constructor. As the constructor is public, one can call it with empty profiles/dependentOn lists, that is forbidden in the parser.

Reply:
The only problem is for version as in the constructor it is an int and Integer in the parsing context to detect missing version

Reply:
Done except for version